File: challenges/30_days/day_20/solution.rb
@n = gets.strip.to_i
@a = gets.strip
@a = @a.split(' ').map(&:to_i)

def swap_round
    i = 0
    @num_swaps = 0
    while i < @n - 1 do
        if @a[i] > @a[i + 1] then
            @a[i], @a[i + 1] = @a[i + 1], @a[i]
            i += 1
            @num_swaps += 1
        else
            i += 1
        end
    end
end

@total_swaps = 0

until @num_swaps == 0 do
    swap_round
    @total_swaps += @num_swaps
end

puts "Array is sorted in #{@total_swaps} swaps."
puts "First Element: #{@a[0]}"
puts "Last Element: #{@a[-1]}"

File: challenges/30_days/day_20/solution.py
import sys

class Solution:
    def __init__(self, n, a):
        self.n = n
        self.a = a
        self.total_swaps = 0
        self.num_swaps = 0

    def swap_round(self):
        self.num_swaps = 0
        for x in range((self.n) - 1):
            if self.a[x] > self.a[x+1]:
                self.a[x], self.a[x+1] = self.a[x+1], self.a[x]
                self.num_swaps += 1

    def bubble_sort(self):
        self.num_swaps = 1
        while self.num_swaps != 0:
            self.swap_round()
            self.total_swaps += self.num_swaps
        else:
            print('Array is sorted in %s swaps.'%(self.total_swaps))
            print('First Element: %s'%(self.a[0]))
            print('Last Element: %s'%(self.a[-1]))